docs/USAGE-GUIDE.md

```diff
@@ -77,35 +77,57 @@ And when invoking CMake for your project, do one of two things:
  1. Set the `CMAKE_INSTALL_PREFIX` variable with the path to your s2n-tls build.
  2. If you have globally installed s2n-tls, do nothing, it will automatically be found.
 
-## Building s2n-tls with OpenSSL-1.1.1
-
-To build s2n-tls with OpenSSL-1.1.1, do the following:
+## Building s2n-tls with Openssl
 
+We keep the build artifacts in the -build directory:
 ```shell
-# We keep the build artifacts in the -build directory
 cd libcrypto-build
+```
 
-# Download the latest version of OpenSSL
-curl -LO https://www.openssl.org/source/openssl-1.1.1-latest.tar.gz
-tar -xzvf openssl-1.1.1-latest.tar.gz
+Download the desired Openssl version:
+```shell
+# Download Openssl 3.0.5
+curl -L -o openssl.tar.gz https://github.com/openssl/openssl/archive/refs/tags/openssl-3.0.5.tar.gz
+tar -xzvf openssl-3.0.5.tar.gz
+cd `tar ztf openssl-3.0.5.tar.gz | head -n1 | cut -f1 -d/`
+
+# Download OpenSSL-1.1.1
+curl -LO https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1.tar.gz
+tar -xzvf OpenSSL_1_1_1.tar.gz
+cd `tar ztf OpenSSL_1_1_1.tar.gz | head -n1 | cut -f1 -d/`
+
+# Download OpenSSL-1.0.2
+curl -LO https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_0_2.tar.gz
+tar -xzvf OpenSSL_1_0_2.tar.gz
+cd `tar ztf OpenSSL_1_0_2.tar.gz | head -n1 | cut -f1 -d/`
+```
 
-# Build openssl libcrypto
-cd `tar ztf openssl-1.1.1-latest.tar.gz | head -n1 | cut -f1 -d/`
+Build Openssl. The following config command disables numerous Openssl features and algorithms which
+are not used by s2n-tls. A minimal feature-set can help prevent exposure to security vulnerabilities.
+```shell
+# When build openssl 1.0.2 libcrypto also include the following flags
+./config ... no-libunbound no-gmp no-jpake no-krb5 no-store ...
+
+# Build Openssl libcrypto
 ./config -fPIC no-shared              \
          no-md2 no-rc5 no-rfc3779 no-sctp no-ssl-trace no-zlib     \
          no-hw no-mdc2 no-seed no-idea enable-ec_nistp_64_gcc_128 no-camellia\
          no-bf no-ripemd no-dsa no-ssl2 no-ssl3 no-capieng                  \
          -DSSL_FORBID_ENULL -DOPENSSL_NO_DTLS1 -DOPENSSL_NO_HEARTBEATS      \
          --prefix=`pwd`/../../libcrypto-root/
```

Comment:
If you're going to make the openssl 1.0.2 config command look real, it probably should be real. Otherwise, I'd just make the list of additional flags just a list, without the "./config".

And like above, I wouldn't make this all one big block of code.